src/thor/triplegbuilder.cc

```diff
@@ -1069,16 +1072,15 @@ TripLeg_Edge* AddTripEdge(const AttributesController& controller,
                           const uint32_t start_node_idx,
                           const bool has_junction_name,
                           const graph_tile_ptr& start_tile,
-                          const uint8_t restrictions_idx,
-                          float elapsed_secs,
-                          bool blind_instructions) {
+                          bool blind_instructions,
+                          EdgeInfo& edgeinfo,
+                          const std::pair<std::vector<std::pair<float, float>>, uint32_t>& levels) {
```

Comment:
pass edgeinfo and levels, and while at it simplify the  signature a bit by simply passing `edge_itr`